jclouds-plugin/src/main/java/jenkins/plugins/jclouds/compute/JCloudsSlaveTemplate.java

```diff
@@ -312,41 +389,27 @@ public NodeMetadata get() {
         }
 
         Statement initStatement = null;
-        Statement bootstrap = null;
 
         if (this.preExistingJenkinsUser) {
             if (this.initScript.length() > 0) {
                 initStatement = Statements.exec(this.initScript);
             }
         } else {
-            // setup the jcloudTemplate to customize the nodeMetadata with jdk, etc. also opening ports
+            // provision jenkins user
             AdminAccess adminAccess = AdminAccess.builder().adminUsername(getJenkinsUser())
-                    .installAdminPrivateKey(installPrivateKey) // some VCS such as Git use SSH authentication
-                    .grantSudoToAdminUser(allowSudo) // no need
-                    .adminPrivateKey(getCloud().privateKey) // temporary due to jclouds bug
-                    .authorizeAdminPublicKey(true).adminPublicKey(getCloud().publicKey).adminHome(getFsRoot()).build();
-
+                .installAdminPrivateKey(installPrivateKey) // some VCS such as Git use SSH authentication
+                .grantSudoToAdminUser(allowSudo) // no need
+                .adminPrivateKey(getJenkinsPrivateKey()) // temporary due to jclouds bug
+                .authorizeAdminPublicKey(true).adminPublicKey(getJenkinsPublicKey()).adminHome(getFsRoot()).build();
```

Comment:
Yeah, I failed to understand what `AdminAccess` is supposed to do and how. Anyway, I wanted to create some end-to-end [ATH](https://github.com/jenkinsci/acceptance-test-harness) tests for JClouds/OpenStack for quite some time. If you do not mind, I suggest not to merge this before we have some coverage.

All in all I like the change. Great work on preserving backward compatibility!